pkg/monitoring/metrics/common/client/latency_adapter.go

```diff
@@ -31,5 +32,21 @@ type latencyAdapter struct {
 }
 
 func (l *latencyAdapter) Observe(_ context.Context, verb string, u url.URL, latency time.Duration) {
+	if verb == "GET" {
+		resource := findResource(u)
+		if resource == "" {
+			resource = "none"
+		}
```

Comment:
I see that you improved a bit, was going to comment on this, but what would you think about:

```
func getVerbLabelValue(u url.URL, verb string) string {
	if verb != "GET" {
		return verb
	}

	if strings.Contains(u.Path, "/watch/") || u.Query().Get("watch") == "true" {
		return "WATCH"
	}

	resource := findResource(u)
	if resource == "" {
		resource = "none"
	}

	if strings.HasSuffix(u.Path, resource) {
		// If the resource is the last element in the url, then
		// we're asking to list all resources of that type instead
		// of getting an individual resource
		return "LIST"
	}

	return verb
}
```

yet a bit better, no?

Reply:
and then just
`	l.m.WithLabelValues(getVerbLabelValue(u, verb), u.String()).Observe(latency.Seconds())`

Reply:
Yes, it's better! The refactoring made the code cleaner and modular. Thanks @machadovilaca for simplifying the logic and separating concerns.